src/table.js

```diff
@@ -543,8 +543,9 @@ export function getTypeValidator(colType) {
 // DuckDBClient for data arrays, too, and then we wouldn’t need our own __table
 // function to do table operations on in-memory data?
 export function __table(source, operations) {
-  const input = source;
   let {schema, columns} = source;
+  if (!schema || !isValidSchema(schema)) source.schema = inferSchema(source);
```

Comment:
This is mutating the source.schema which will be visible externally and we should avoid because mutation of inputs is an unexpected side-effect of calling this function.

If necessary we can use a WeakMap to instead associate sources with valid schemas if we don’t want to re-infer them repeatedly. (It might also make sense to move this schema inference earlier, say in loadDataSource? Not sure though.)